Source/DafnyCore/Compilers/SinglePassCompiler.cs

```diff
@@ -2740,7 +2740,7 @@ void TrCasePatternOpt<VT>(CasePattern<VT> pat, Expression rhs, Action<ConcreteSy
       }
     }
 
-    void TrExprOpt(Expression expr, Type resultType, ConcreteSyntaxTree wr, IVariable/*?*/ accumulatorVar) {
+    void TrExprOpt(Expression expr, Type resultType, ConcreteSyntaxTree wr, bool inLetExprBody, [CanBeNull] IVariable accumulatorVar) {
```

Comment:
I think this and TrCaseExpressionOpt deserve a short comment about what optimizations they are applying and when they are valid. AFAICT it's something like "parallel optimized implementations for when we are translating the return value of a function, and hence can do things like compile chains of LetExprs into sequences of statements"